File: java/src/main/java/kkurczewski/deser/value/ValueObjectController.java
package kkurczewski.deser.value;

import kkurczewski.deser.value.dto.Wrapper;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ValueObjectController {

    @PostMapping("/wrapper")
    public Wrapper wrapper(@RequestBody Wrapper wrapper) {
        return wrapper;
    }
}

File: kotlin/src/main/kotlin/pl/kkurczewski/deser/value/ValueObjectController.kt
package pl.kkurczewski.deser.value

import pl.kkurczewski.deser.value.dto.Wrapper
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class ValueObjectController {
    @PostMapping("/wrapper")
    fun wrapper(@RequestBody wrapper: Wrapper) = wrapper
}